File: be_fastapi/app/models/zone.py
from sqlalchemy import Column, String, TIMESTAMP, ForeignKey, text
from sqlalchemy.dialects.postgresql import UUID, JSONB
from .base import Base

class Zone(Base):
    __tablename__ = "zones"

    id = Column(UUID(as_uuid=True), primary_key=True, server_default=text("uuidv7()"))
    name = Column(String, nullable=False)

    source_id = Column(
        UUID(as_uuid=True),
        ForeignKey("sources.id", ondelete="CASCADE"),
        nullable=False, 
        index=True
    )

    coordinates = Column(JSONB, nullable=False)

    created_at = Column(TIMESTAMP(timezone=True), nullable=False, server_default=text("NOW()"))
    updated_at = Column(TIMESTAMP(timezone=True), nullable=False, server_default=text("NOW()"), onupdate=text("NOW()"))

File: fe_react/src/types/zone.ts
export interface Zone {
  id: string;
  name: string;
  source_id: string;
  coordinates: { points: [number, number][] };
  created_at: string;
  updated_at: string;
}

export interface CreateZonePayload {
  name: string;
  source_id: string;
  coordinates: { points: [number, number][] };
}

export interface UpdateZonePayload {
  name: string;
  coordinates: { points: [number, number][] };
}
